### PR TITLE
Fix PID generation in some edge cases (#88)

### PR DESCRIPTION
**Pull request by Real96**
Fix wrong PID generation in some Raid Shiny cases (#88)

### CODE REVIEW

File: Core/Generator/RaidGenerator.cpp [hunk 1]
```diff
@@ -62,12 +62,13 @@ std::vector<State> RaidGenerator::generate(const StateFilter &filter, u64 seed)
             // PID is later modified using the actual TID/SID of trainer if necessary
             u16 shinyXor = (sidtid >> 16) ^ (sidtid & 0xffff) ^ (pid >> 16) ^ (pid & 0xffff);
             u16 psv = ((pid >> 16) ^ (pid & 0xffff)) >> 4;
+            u8 PIDShinyType = (pid >> 16) ^ (pid & 0xffff) ^ tid ^ sid;
```
Comment (Admiral-Fish) [inline Core/Generator/RaidGenerator.cpp]: Why is this u8 and not u16?

Comment (Admiral-Fish) [inline Core/Generator/RaidGenerator.cpp]: Also should probably change name to realXor as we have discussed.

File: Core/Generator/RaidGenerator.cpp [hunk 2]
```diff
@@ -62,12 +62,13 @@ std::vector<State> RaidGenerator::generate(const StateFilter &filter, u64 seed)
             // PID is later modified using the actual TID/SID of trainer if necessary
             u16 shinyXor = (sidtid >> 16) ^ (sidtid & 0xffff) ^ (pid >> 16) ^ (pid & 0xffff);
```
Comment (Admiral-Fish) [inline Core/Generator/RaidGenerator.cpp]: Change this name to fakeXor as we have discussed.

File: Core/Generator/RaidGenerator.cpp [hunk 3]
```diff
@@ -93,26 +94,13 @@ std::vector<State> RaidGenerator::generate(const StateFilter &filter, u64 seed)
         }
         else // Force shiny
         {
-            u16 shinyXor = (sidtid >> 16) ^ (sidtid & 0xffff) ^ (pid >> 16) ^ (pid & 0xffff);
-            if (shinyXor >= 16) // Check if PID is not normally shiny
+            result.setShiny(2);
+            u8 PIDShinyType = (pid >> 16) ^ (pid & 0xffff) ^ tid ^ sid;
```
Comment (Admiral-Fish) [inline Core/Generator/RaidGenerator.cpp]: Same questions as above

File: Core/Generator/RaidGenerator.cpp [hunk 4]
```diff
@@ -93,26 +94,13 @@ std::vector<State> RaidGenerator::generate(const StateFilter &filter, u64 seed)
...
+            if (PIDShinyType >= 16 || PIDShinyType) // Check if PID is not normally shiny
```
Comment (Admiral-Fish) [inline Core/Generator/RaidGenerator.cpp]: I don't understand this conditional. This is really just checking PIDShinyType != 0

File: Core/Generator/RaidGenerator.cpp (hunk 1 above)
Comment (Real96) [reply Core/Generator/RaidGenerator.cpp]: Fixed

Comment (Real96) [reply Core/Generator/RaidGenerator.cpp]: Done

File: Core/Generator/RaidGenerator.cpp (hunk 2 above)
Comment (Real96) [reply Core/Generator/RaidGenerator.cpp]: Done

File: Core/Generator/RaidGenerator.cpp (hunk 3 above)
Comment (Real96) [reply Core/Generator/RaidGenerator.cpp]: Done

File: Core/Generator/RaidGenerator.cpp (hunk 4 above)
Comment (Real96) [reply Core/Generator/RaidGenerator.cpp]: Fixed

**Pull request merged**